src/ompl/util/src/RandomNumbers.cpp

```diff
@@ -217,3 +238,297 @@ void ompl::RNG::eulerRPY(double value[3])
+    boost::variate_generator<boost::mt19937&, boost::uniform_on_sphere<double, container_type_t> > normalizedVector(generator_, uniformSphereDist);
+
+    // Create a uBLAS-vector-view of the C-style array without copying data
+    container_type_t rVector(n, value);
+
+    // Generate a random value, the variate_generator is returning a shallow_array_adaptor, which will modify the value array:
+    rVector = normalizedVector();
+}
+
+// See: http://math.stackexchange.com/a/87238
+void ompl::RNG::uniformInBall(double r, unsigned int n, double value[])
+{
+    // Draw a random point on the unit sphere
+    this->uniformNormalVector(n, value);
+
+    // Draw a random radius scale
+    double radiusScale = r*std::pow(this->uniformReal(0.0, 1.0), 1.0/static_cast<double>(n));
+
+    // Scale the point on the unit sphere
+    for (unsigned int i = 0u; i < n; ++i)
+    {
+        value[i] = radiusScale * value[i];
+    }
+}
+
+void ompl::RNG::uniformProlateHyperspheroidSurface(ProlateHyperspheroidPtr phsPtr, unsigned int n, double value[])
+{
+    // Variables
+    // The spherical point as a std::vector
+    std::vector<double> sphere(n);
+
+    // Get a random point on the sphere
+    this->uniformNormalVector(n, &sphere[0]);
+
+    // Transform to the PHS
+    phsPtr->transform(n, &sphere[0], value);
+}
+
+void ompl::RNG::uniformProlateHyperspheroid(ProlateHyperspheroidPtr phsPtr, unsigned int n, double value[])
+{
+    // Variables
+    // The spherical point as a std::vector
+    std::vector<double> sphere(n);
+
+    // Get a random point in the sphere
+    this->uniformInBall(1.0, n, &sphere[0]);
+
+    // Transform to the PHS
+    phsPtr->transform(n, &sphere[0], value);
+}
+
+
+
+
+struct ompl::ProlateHyperspheroid::phsData
+{
+    /** \brief The dimension of the prolate hyperspheroid.*/
+    unsigned int dim_;
+    /** \brief The minimum possible transverse diameter of the PHS. Defined as the distance between the two foci*/
+    double minTransverseDiameter_;
+    /** \brief The transverse diameter of the PHS. */
+    double transverseDiameter_;
+    /** \brief The measure of the PHS. */
+    double phsMeasure_;
+    /** \brief The first focus of the PHS (i.e., the start state of the planning problem)*/
+    Eigen::VectorXd xFocus1_;
+    /** \brief The second focus of the PHS (i.e., the goal state of the planning problem)*/
+    Eigen::VectorXd xFocus2_;
+    /** \brief The centre of the PHS. Defined as the average of the foci.*/
+    Eigen::VectorXd xCentre_;
+    /** \brief The rotation from PHS-frame to world frame. Is only calculated on construction. */
+    Eigen::MatrixXd rotationWorldFromEllipse_;
+    /** \brief The transformation from PHS-frame to world frame. Is calculated every time the transverse diameter changes. */
+    Eigen::MatrixXd transformationWorldFromEllipse_;
+    /** \brief Whether the transformation is up to date */
+    bool isTransformUpToDate_;
```

Comment:
For memory alignment considerations I would move this bool next to the uint at the top.